src/lib/util/util.js

```diff
@@ -141,6 +141,26 @@ class Util {
 		return source;
 	}
 
+	/**
+	 * Deep freeze a value
+	 * @since 0.5.0
+	 * @param {*} source The object to freeze
+	 * @returns {Readonly<*>}
+	 */
+	static deepFreeze(source) {
+		if (typeof source === 'object' && source !== null) {
+			Object.freeze(source);
+			if (Array.isArray(source)) {
+				for (const value of source) Util.deepFreeze(value);
+			} else {
```

Comment:
This function will not work on maps or sets. Their keys/values won't be frozen.

Reply:
Fixed in 02d6353